ludwig/automl/automl.py

```diff
@@ -144,6 +157,65 @@ def auto_train(
     return train_with_config(dataset, config, output_directory=output_directory, random_seed=random_seed, **kwargs)
 
 
+def create_auto_config_with_dataset_profile(
```

Comment:
nit: Can we add the PublicAPI annotation?

Reply:
It probably makes sense to leave this unannotated as I'd like this API to harden a bit before we commit to maintaining it in public.

Reply:
Got it! Maybe in that case, we can flag it as DeveloperAPI for now and then make it PublicAPI later?

Reply:
SGTM!